sound/soc/sof/intel/Makefile

```diff
@@ -12,6 +12,10 @@ snd-sof-intel-hda-common-objs := hda.o hda-loader.o hda-stream.o hda-trace.o \
 				 hda-dai.o hda-bus.o \
 				 apl.o cnl.o
 
+ifdef CONFIG_SND_SOC_SOF_DEBUG_PROBES
+	snd-sof-intel-hda-common-objs += probe.o
+endif
```

Comment:
there should be a core part and an intel-specific implementation. The abstraction is missing.

Reply:
Discussions, discussion..
Time for a discussion!